src/features/element-hiding.js

```diff
@@ -334,4 +297,43 @@ export default class ElementHiding extends ContentFeature {
             applyRules(activeRules)
         })
     }
+
+    /**
+     * Apply relevant hiding rules to page at set intervals
+     * @param {Object[]} rules
+     * @param {string} rules[].selector
+     * @param {string} rules[].type
+     */
+    applyRules (rules) {
+        this.addDebugFlag()
```

Comment:
This will always fire when we have hide rules.

I would suggest moving it to `collapseDomNode` instead and leaving this method.